dotcom-rendering/package.json

```diff
@@ -240,7 +249,8 @@
 		"webpack-merge": "^5.7.3",
 		"webpack-messages": "^2.0.4",
 		"webpack-node-externals": "^3.0.0",
-		"webpack-sources": "^3.2.3"
+		"webpack-sources": "^3.2.3",
+		"zod": "^3.20.2"
```

Comment:
Are we bloating the deployed app's `node_modules` for the sake of a tool that is adjacent, but not directly part of it.

A slight concern as we have filled the disk on our instances that we run this on in prod. This  feels indirect rather than explicitly part of the app.

I know we veered away from it - but maybe it's time to bring back `devDependencies` and use `install --prod` when we're building the webapp?

Or are these able to maybe live in the root `/scripts` folder and have their own `/package.json` or use `deno`?

Reply:
I wrote this in deno originally, but there’s no support for it in TeamCity, so this was the next best thing. The only way to ensure that things are actually reprensenting the correct state of things is having them run as part of the build step.

To give some perspective, we consider `webpack-bundle-analyser` to be a worthy addition at 243kB, but these addtions combined are considerably smaller at 141 kB =  [`zod` (50 kB)](https://bundlephobia.com/package/zod@3.20.2) + [`unified` (11 kB)](https://bundlephobia.com/package/unified@10.1.2) + [`remark-parse` (50 kB)](https://bundlephobia.com/package/remark-parse@10.0.1) + [`rehype-stringify` (30 kB)](https://bundlephobia.com/package/rehype-stringify@9.0.3)

Reply:
Using `devDependencies` would not address the size of our node modules, as we need this script to run in TeamCity to create the output to be uploaded to Riff Raff. I think embedding all dependencies inside our server and not shipping node modules would be more advisable.

As you said, I don’t think any of this is a blocker to this PR.